### PR TITLE
chore(workflow): allow renovate to bump npm dependencies

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

- Allow renovate to bump npm dependencies by removing `"enabled": false`.
- Bump major in a separate PR by adding `"matchUpdateTypes": ["patch", "minor"]`.
- Let renovate always bump package.json by adding `"rangeStrategy": "bump"`, avoid only updating lock files (such as https://github.com/web-infra-dev/rspack/pull/7633)

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -116,16 +116,15 @@
...
-        "@playwright/test",
-        "react",
-        "react-dom",
         "typescript",
         "pnpm"
-      ]
+      ],
+      "rangeStrategy": "bump",
+      // bump major in a separate PR
+			"matchUpdateTypes": ["patch", "minor"]
```
Comment (SoonIter) [inline .github/renovate.json5:127]: should use spaces for text-indent

**Comment by SoonIter** [comment]
LGTM

I think it is related to https://github.com/web-infra-dev/rspack/issues/7156 , I had an idea at that time to distinguish the renovate group to  achieve a regular cycle

1. "easy-merge-daily" group like patch or minor change of `sass` `file-loader` `less-loader` `postcss-loader` `webpack` etc

2. "hard-merge-weekly-or-monthly" group like major version of`swc` `react` `sass-embed` `skykv` `rspack-resolver` etc

this is my ultimate goal, I'm not sure if it can be achieved. At present, I haven't merged regularly.

**Comment by chenjiahan** [comment]
Nice, for the `easy-merge` groups, I prefer weekly upgrades, because we do not want to approve the PR every day.

For the `hard-merge` groups, each one should be handled separately,  as they usually require changes.

File: .github/renovate.json5 (hunk 1 above)
Comment (chenjiahan) [reply .github/renovate.json5]: fixed

**Review by SoonIter — APPROVED** [review]

**Pull request merged**